From 20b849478bab7f23d0b8f8b096c1bdc3137dfbfd Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 22 Apr 2021 19:46:29 +0300
Subject: [PATCH] [SW-28697] module_lkd: clear comms status bit

Since we do not know if the loaded FW which intends to run supports the
new comms protocol, the most safe way is to clear comms status bit and
let the loaded FW set it if supported.

Change-Id: I75f179667508facbf9fd1fe1f10756f493e1ceec
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 2b94d7d..a2e37f3 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -458,10 +458,17 @@
 
 static void module_lkd_run_noret_exec(void)
 {
+	/*
+	 * List of status bits we clear because we do not know if the FW we are
+	 * loading supports these functionalities
+	 */
+	u32 sts_msk = CPU_BOOT_DEV_STS0_FW_HARD_RST_EN |
+		CPU_BOOT_DEV_STS0_FW_LD_COM_EN;
+
 	preboot_security_cleanup_before_run();
 	hl_ts_shutdown();
-	/* We do not know if the FW we are loading supports hard rst */
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN, false);
+	cpu_boot_dev_sts_clrset(sts_msk, false);
+
 	fit_run_image();
 }
 
